estuary/ui
# Default the data-plane selector to GCP data-plane option

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/1272

## Changes

### 1272

The following features are included in this PR:

* Fetch data plane options in details form store hydrate action.

* Default the data-plane selector to the GCP data-plane option (i.e., `gcp-us-central1-c1`) in production, create workflows.

* Default the data-plane selector to the single, out-of-the-box data-plane option (i.e., `local-cluster`) in local, create workflows.

* Display an error alert when the details form store fails to hydrate and `hydrationError` has a value.

* Store the suffix of the default data-plane option in an environment variable.

* Add a custom LogRocket event to log when the data-plane option corresponding to the `data_plane_id` in the URL cannot be found.

* Move `getDataPlaneScope` and `parseDataPlaneName` into the data-plane utils file.

* Create a shared constant for the public data-plane prefix (i.e., `ops/dp/public/`).

* Minorly refactor the `parseDataPlaneName` util.

## Tests

### Manually tested

Approaches to testing are as follows:

* Validate that the data-plane selector is defaulted to the GCP data-plane option (i.e., `gcp-us-central1-c1`) in production, create workflows.

* Validate that the data-plane selector is defaulted to the single, out-of-the-box data-plane option (i.e., `local-cluster`) in local, create workflows.

* Validate that the default value of the data-plane selector can be changed and retained when the page is refreshed in production and local, create workflows.

* Validate that the data-plane selector is defaulted to the option corresponding to the data plane ID in the URL in local, edit workflows.

* Validate that a LogRocket event is created when the data-plane option corresponding to the data plane ID in the URL is not found when hydrating local workflows.

* Validate that an error alert is displayed when data-plane options fail to be fetched or none are found in local workflows.

### Automated tests

Test coverage added for the following utils:

* `getDataPlaneScope`

* `parseDataPlaneName`

#### Playwright tests ran locally


## Screenshots

### Production

**Capture create workflow**

<img width="525" alt="pr_screenshot-1242-default_data_plane_selector-prod-capture_create" src="https://github.com/user-attachments/assets/e92bfe9d-43ed-4581-ac36-5c97ca6a2606">

<br />
<br />

**Materialization create workflow**

<img width="524" alt="pr_screenshot-1242-default_data_plane_selector-prod-materialization_create" src="https://github.com/user-attachments/assets/3ebd9e13-1366-4f05-982f-4073ffd98cdd">
